src/CLI/Command.php

```diff
@@ -17,4 +17,59 @@ public function register() {
 	abstract protected function description();
 	abstract protected function arguments();
 
+	/**
+	 * Frees up memory for long running processes.
+	 *
+	 * @return void
+	 */
+	protected function stop_the_insanity() : void {
+		global $wpdb, $wp_actions, $wp_filter, $wp_object_cache;
+
+		// Reset queries.
+		$wpdb->queries = [];
+
+		// Prevent wp_actions from growing out of control.
+
+		$wp_actions = []; // @codingStandardsIgnoreLine
+		if ( is_object( $wp_object_cache ) ) {
+			$wp_object_cache->group_ops      = [];
+			$wp_object_cache->stats          = [];
+			$wp_object_cache->memcache_debug = [];
+			$wp_object_cache->cache          = [];
+			if ( method_exists( $wp_object_cache, '__remoteset' ) ) {
+				$wp_object_cache->__remoteset();
+			}
+		}
+
+		/**
+		 * The WP_Query class hooks a reference to one of its own methods
+		 * onto filters if update_post_term_cache or
+		 * update_post_meta_cache are true, which prevents PHP's garbage
+		 * collector from cleaning up the WP_Query instance on long-
+		 * running processes.
+		 *
+		 * By manually removing these callbacks (often created by things
+		 * like get_posts()), we're able to properly unallocate memory
+		 * once occupied by a WP_Query object.
+		 */
+		if ( isset( $wp_filter['get_term_metadata'] ) ) {
+			/*
+			 * WordPress 4.7 has a new Hook infrastructure, so we need to make sure
+			 * we're accessing the global array properly.
+			 */
+			if ( class_exists( 'WP_Hook' ) && $wp_filter['get_term_metadata'] instanceof \WP_Hook ) {
+				$filter_callbacks = &$wp_filter['get_term_metadata']->callbacks;
+			} else {
+				$filter_callbacks = &$wp_filter['get_term_metadata'];
```

Comment:
For our purposes, we could get rid of the pre-4.7 check and just use `$wp_filter['get_term_metadata']->callbacks`